python/packages/ai/teams/ai/ai.py

```diff
@@ -336,3 +336,29 @@ async def _on_too_many_steps(
     ) -> str:
         self._logger.error("The run retrieval for the Assistants Planner has expired.")
         return ActionTypes.STOP
+
+    async def do_action(
```

Comment:
for testing, we don't yet have a `test_ai.py` class but this would be the perfect opp to start one! you can look at the other test files to see how we mock out state + context

Reply:
should I attach that to this work, or should a separate issue / pr be made? What is the standard here?

Reply:
can attach to this PR, test updates accompany corresponding file changes